src/TermAuctionList.sol

```diff
@@ -206,6 +206,8 @@ library TermAuctionList {
                 listData.nodes[prev].next = next;
                 delete listData.nodes[current];
                 delete listData.offers[current];
+                current = prev;
+

```

Comment:
I suggest the following optimization to avoid useless assignments:
```solidity
    if (removeNode) {
        // Update the list to remove the current node
        delete listData.nodes[current];
        delete listData.offers[current];
        if (current == listData.head) {
            listData.head = next;
        } else {
            listData.nodes[prev].next = next;
            current = prev;
        }
   }
```